### DISCUSSION TITLE
account_activity_times error

### DISCUSSION THREAD

**Discussion by a14guguliye** [janko/rodauth-rails#146] [Q&A] [answered]
Hello,
why the account_activity_times table is not generated by rodauth automatically although all others are being generated? Meanwhile, the below code gives me the error. I know it is not an issue but trying to solve it below for weeks now, but can not solve it. Can someone please help?
class New < ActiveRecord::Migration[7.0] def change create_table(:account_activity_times) do foreign_key :id, :account, :primary_key=>true, :type=>:Bignum DateTime :last_activity_at, :null=>false DateTime :last_login_at, :null=>false DateTime :expired_at end end end

**Comment by janko** [comment, marked as answer]
It's not true that tables for all other features are generated (e.g. MFA tables aren't), the install generator creates tables only for features that are enabled by default. Enabling all features would be overwhelming for the user, and wouldn't be a good default, and as a result it doesn't make sense to generate tables for unused features.

Your migration code doesn't work because you've copied Sequel migration syntax, which differs greatly from Active Record's. The migration generator will create the table using Active Record syntax:

```sh
$ rails g rodauth:migration account_expiration
$ rails db:migrate
```

**Reply by a14guguliye** [reply]
Thank u very much for ur answer. Has it been specified in the documentation? I want to see where i miss the things

**Reply by janko** [reply]
It's specified here: https://github.com/janko/rodauth-rails#migrations
